cmd/otel-allocator/config/config.go

```diff
@@ -2,31 +2,54 @@ package config
 
 import (
 	"errors"
+	"flag"
 	"fmt"
+	"io/fs"
 	"io/ioutil"
+	"path/filepath"
+	"time"
 
+	"github.com/go-logr/logr"
 	promconfig "github.com/prometheus/prometheus/config"
 	_ "github.com/prometheus/prometheus/discovery/install"
+	"github.com/spf13/pflag"
 	"gopkg.in/yaml.v2"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
 // ErrInvalidYAML represents an error in the format of the original YAML configuration file.
 var (
 	ErrInvalidYAML = errors.New("couldn't parse the loadbalancer configuration")
 )
 
-const defaultConfigFile string = "/conf/targetallocator.yaml"
+const DefaultResyncTime = 5 * time.Minute
+const DefaultConfigFilePath string = "/conf/targetallocator.yaml"
 
 type Config struct {
 	LabelSelector map[string]string  `yaml:"label_selector,omitempty"`
 	Config        *promconfig.Config `yaml:"config"`
 }
 
-func Load(file string) (Config, error) {
-	if file == "" {
-		file = defaultConfigFile
-	}
+type PrometheusCRWatcherConfig struct {
+	Enabled *bool
+}
 
+type CLIConfig struct {
+	ListenAddr     *string
+	ConfigFilePath *string
+	ClusterConfig  *rest.Config
+	// KubeConfigFilePath empty if in cluster configuration is in use
+	KubeConfigFilePath string
+	RootLogger         logr.Logger
+	PromCRWatcherConf  PrometheusCRWatcherConfig
+}
+
+func Load(file string) (Config, error) {
 	var cfg Config
 	if err := unmarshal(&cfg, file); err != nil {
```

Comment:
question: It seems previously the allocator would load a default config file if no file was provided, it seems that was because we would always pass in an empty string in `main.go`. Should that functionality remain in case there any consumers of this function?

Reply:
Are you referring to users of the TargetAllocator or Devs which import this package? 

In the first case the behavior has not changed. It is still loading the same default file as before. This is implement using pflags now. https://github.com/open-telemetry/opentelemetry-operator/pull/653/files#diff-d292758014333553bcd82dfdf75743ef82b0b32f6414e2dc345777fc342e4480R77